src/global-types.ts

```diff
@@ -59,6 +73,108 @@ declare global {
           ErrorsShape
         >,
       ) => FieldRef<unknown>;
+
+      effectConnection: 'relay' extends PluginName ? <
+          Type extends OutputType<Types>,
+          Nullable extends boolean,
+          ResolveReturnShape,
+          // Effect Types:
+          ServiceEntriesShape extends readonly [...EffectPluginTypes.ServiceEntry[]],
+          ContextsShape extends readonly [...EffectPluginTypes.Context[]],
+          LayersShape extends readonly [...EffectPluginTypes.Layer[]],
+          ErrorsShape extends readonly [...any[]],
+          // Relay Types:
+          Args extends InputFieldMap = {},
+          ConnectionInterfaces extends InterfaceParam<Types>[] = [],
+          EdgeInterfaces extends InterfaceParam<Types>[] = [],
+        >(
+          options: EffectPluginTypes.ConnectionFieldOptions<
+            Types,
+            ParentShape,
+            Type,
+            Args,
+            Nullable,
+            ResolveReturnShape,
+            ServiceEntriesShape,
+            ContextsShape,
+            LayersShape,
+            ErrorsShape,
+            Kind
+          >,
+          ...args: NormalizeArgs<
+            [
+              connectionOptions:
+                | ConnectionObjectOptions<
+                  Types,
+                  Type,
+                  false,
+                  false,
+                  ResolveReturnShape,
+                  ConnectionInterfaces
+                >
+                | ObjectRef<
+                  EffectPluginTypes.ShapeFromConnection<
+                    ConnectionShapeHelper<Types, ShapeFromTypeParam<Types, Type, false>, false>
+                  >
+                >,
+              edgeOptions:
+                | ConnectionEdgeObjectOptions<
+                  Types,
+                  Type,
+                  false,
+                  ResolveReturnShape,
+                  EdgeInterfaces
+                >
+                | ObjectRef<{
+                  cursor: string;
+                  node?: ShapeFromTypeParam<Types, Type, false> | null | undefined;
+                }>,
+            ],
+            0
+          >
+        ) => FieldRef<
+          EffectPluginTypes.ShapeFromConnection<
+            ConnectionShapeHelper<Types, ShapeFromTypeParam<Types, Type, false>, Nullable>
+          >
+        >
+        : '@pothos/plugin-relay is required to use this method';
+    }
+
+    export interface ConnectionShapeHelper<Types extends SchemaTypes, T, Nullable> {}
```

Comment:
These are defined in the global namespace.  This allows this plugin to reference the types without fully re-implementing them.  for example `ConnectionFieldOptions` is empty here, but when the relay plugin is loaded, it will be extended with things like `maxSize` and other connection specific options

Reply:
`ConnectionShapeHelper` is especially weird, it basically will be implemented as something like:

```ts
`interface ConnectionShapeHelper<Types extends SchemaTypes, T, Nullable>  {
  shape: ConnectionShape<Types, T, Nullable>
```
this in combination with this type
```
export type ShapeFromConnection<T> = T extends { shape: unknown } ? T['shape'] : never;
```

allows this plugin to get the shape of a connection 

roughly `{ pageInfo: PageInfo, edges: { cursor: string; node: T }[] }` (actual types are a bit more complicated to handle nullability/promises etc)

Reply:
the overall goal is to limit what plugins know about the details of Connections while still being fully compatible